src/tree.rs

```diff
@@ -230,28 +673,142 @@ struct Entry {
 pub struct Node<'t>(&'t Tree, &'t Entry);
 
 impl<'t> Node<'t> {
+    /// Returns an iterator for all resolved children of this node, including
+    /// reparented orphans.
     pub fn children<'n>(&'n self) -> impl Iterator<Item = Node<'t>> + 'n {
-        self.1
-            .child_indices
+        let orphans = self.tree().orphan_indices_by_parent_guid.get(&self.entry().item.guid)
             .iter()
-            .map(move |index| self.0.node(*index))
+            .flat_map(|orphan_indices| orphan_indices.iter())
+            .collect::<Vec<_>>();
+
+        let default_orphans = if self.is_default_parent_for_orphans() {
+            self.tree().orphan_indices_by_parent_guid.iter()
+                .filter(|(guid, _)| !self.tree().entry_index_by_guid.contains_key(guid))
+                .flat_map(|(_, child_indices)| child_indices)
+                .collect()
+        } else {
+            Vec::new()
+        };
+
+        self.entry().child_indices.iter()
+            .chain(orphans.into_iter())
```

Comment:
@thomcc I thought about it some more, and I'm unsure how to avoid the allocation here. `self.tree().orphan_indices_by_parent_guid` borrow `self` (and `default_orphans` also borrows `self` in the closure passed to `filter`), so the returned iterator outlives the temporary...and we also move `self` into `filter_map`'s closure.

But, as you said, orphans should be rare...and we already do much more work for diverged nodes, anyway. An extra heap allocation is likely the least of that.